shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -388,10 +388,9 @@ - (void)surfaceUpdated:(BOOL)appeared {
 
 - (void)viewWillAppear:(BOOL)animated {
   TRACE_EVENT0("flutter", "viewWillAppear");
-
-  if (_engineNeedsLaunch) {
+  if (_isEmbedEngine) {
+    [_engine.get() createShell:nil libraryURI:nil];
```

Comment:
I'm pretty sure this will lead to unexpected loss of state.  If we allow the engine to get torn down here, restarting it will prevent th ecrash but also restart the app from scratch.